src/main/java/org/openrefine/extensions/commons/utils/CommonsImportingController.java

```diff
@@ -182,9 +183,124 @@ private void doParsePreview(
+    }
+
+    static public void parse(
+            Project project,
+            ProjectMetadata metadata,
+            final ImportingJob job,
+            String cmtitle,
+            int limit,
+            ObjectNode options,
+            List<Exception> exceptions) throws IOException {
+
+            String pageName = cmtitle;
+            String fileSource = pageName;//FIXME: add filename
+
+            setProgress(job, fileSource, 0);
+
+            TabularImportingParserBase.readTable(
+                    project,
+                    job,
+                    new FilesBatchRowReader(job, fileSource, cmtitle),
+                    limit,
+                    options,
+                    exceptions);
+            setProgress(job, fileSource, 100);
+        }
+
+        static private void setProgress(ImportingJob job, String fileSource, int percent) {
+            job.setProgress(percent, "Reading " + fileSource);
+        }
+
+        static private class FilesBatchRowReader implements TableDataReader {
+            final ImportingJob job;
+            final String fileSource;
+            JsonNode files;
+            String cmtitle;
+            String cmcontinue;
+            String urlBase;
+            private int indexRow = 0;
+            List<Object> rowsOfCells;
+
+            public FilesBatchRowReader(ImportingJob job, String fileSource, String cmtitle) throws IOException {
+
+                this.job = job;
+                this.fileSource = fileSource;
+                this.cmtitle = cmtitle;
+                setURL();
+                getFiles(urlBase);
+
+            }
+            public void setURL() {
+
+                urlBase = "https://commons.wikimedia.org./w/api.php"
+                    + "?action=query&list=categorymembers&cmtitle="
+                    + cmtitle + "&cmtype=file&cmprop=title|type|ids&cmlimit=500&format=json";
+
+            }
+
+            public void getFiles(String urlContinue) throws IOException {
+
+                OkHttpClient client = new OkHttpClient.Builder().build();
+                Request request = new Request.Builder().url(urlContinue).build();
+                Response response = client.newCall(request).execute();
+                JsonNode jsonNode = new ObjectMapper().readTree(response.body().string());
+                cmcontinue = jsonNode.path("continue").path("cmcontinue").asText();
+                files = jsonNode.path("query").path("categorymembers");
+
+            }
+
+            @Override
+            public List<Object> getNextRowOfCells() throws IOException {
+
+                if (files.size() > 0) {
+                    setProgress(job, fileSource, 100 * indexRow / files.size());
+                } else if (indexRow == files.size()) {
+                    setProgress(job, fileSource, 100);
+                }
+
+                if (indexRow == files.size() && !cmcontinue.isBlank()) {
+                    String urlContinue = urlBase + "&cmcontinue=" + cmcontinue;
```

Comment:
Same here: the `cmcontinue` parameter should be added to whatever object holds the GET parameters for this request, instead of appending it to the URL.